app/bundles/ReportBundle/Tests/Event/ReportGeneratorEventTest.php

```diff
@@ -285,4 +289,32 @@ public function testAddCompanyLeftJoinOnlyOnceWhenTableAlreadyJoined(): void
 
         $this->reportGeneratorEvent->addCompanyLeftJoin($this->queryBuilder, ReportGeneratorEvent::COMPANY_PREFIX);
     }
+
+    public function testApplyTagFilter(): void
+    {
+        $connection           = $this->createMock(Connection::class);
+        $connection->method('createQueryBuilder')->willReturn(new QueryBuilder($connection));
+        $connection->method('getExpressionBuilder')->willReturn(new ExpressionBuilder($connection));
+        $this->queryBuilder->method('getConnection')->willReturn($connection);
+
+        $groupExpr = new CompositeExpression(CompositeExpression::TYPE_AND);
+        $filters   = [
+            [
+                'column'    => 'tag',
+                'glue'      => 'and',
+                'value'     => ['1', '2'],
+                'condition' => 'in',
+            ],
+            [
+                'column'    => 'tag',
+                'glue'      => 'and',
+                'value'     => ['3'],
+                'condition' => 'notIn',
+            ],
+        ];
+
+        $this->reportGeneratorEvent->applyTagFilter($groupExpr, $filters[0]);
+        $this->reportGeneratorEvent->applyTagFilter($groupExpr, $filters[1]);
+        Assert::assertSame('(l.id IN (SELECT DISTINCT lead_id FROM lead_tags_xref ltx WHERE ltx.tag_id IN (1, 2))) AND (l.id NOT IN (SELECT DISTINCT lead_id FROM lead_tags_xref ltx WHERE ltx.tag_id IN (3)))', $groupExpr->__toString());
```

Comment:
@patrykgruszka please check the failing CI check that popped up once I merged fresh 5.x branch. For example there are missing table prefixes in the test queries.

Reply:
@escopecz fixed, it should be fine now.